contexts/PaginationContext.tsx

```diff
@@ -0,0 +1,104 @@
+import { createContext, ReactNode, useState } from "react";
+import { PageStatesT, PaginateForT } from "types";
+
+export interface PaginationContextState {
+  pageStates: PageStatesT;
+  goToPage: (paginateFor: PaginateForT, page: number) => void;
+  nextPage: (paginateFor: PaginateForT) => void;
+  previousPage: (paginateFor: PaginateForT) => void;
+  firstPage: (paginateFor: PaginateForT) => void;
+  lastPage: (paginateFor: PaginateForT, lastPageNumber: number) => void;
+  setResultsPerPage: (paginateFor: PaginateForT, resultsPerPage: number) => void;
+}
+
+export const defaultPageState = {
+  pageNumber: 1,
+  resultsPerPage: 10,
+};
+
+export const defaultPageStates = {
+  activeVotesPage: defaultPageState,
+  upcomingVotesPage: defaultPageState,
+  pastVotesPage: defaultPageState,
+  voteHistoryPage: defaultPageState,
+};
+
+export const defaultPaginationContextState: PaginationContextState = {
+  pageStates: defaultPageStates,
+  goToPage: () => null,
```

Comment:
if we dont use a functions return value i think you should use `void`
> The intent of void is that a function's return value will not be observed. This is very different from will be undefined

https://stackoverflow.com/questions/58885485/why-does-typescript-have-both-void-and-undefined